Spanner/src/Transaction.php

```diff
@@ -89,21 +89,28 @@ class Transaction implements TransactionalReadInterface
      * @param bool $isRetry Whether the transaction will automatically retry or not.
      * @param string $tag A transaction tag. Requests made using this transaction will
      *        use this as the transaction tag.
+     * @param array $options [optional] {
+     *     Configuration Options.
+     *
+     *     @type array $begin The begin Transaction options.
+     *           [Refer](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#transactionoptions)
+     * }
      * @throws \InvalidArgumentException if a tag is specified on a single-use transaction.
      */
     public function __construct(
         Operation $operation,
         Session $session,
         $transactionId = null,
         $isRetry = false,
-        $tag = null
+        $tag = null,
+        $options = []
     ) {
         $this->operation = $operation;
         $this->session = $session;
         $this->transactionId = $transactionId;
         $this->isRetry = $isRetry;
 
-        $this->type = $transactionId
+        $this->type = ($transactionId || isset($options['begin']))
```

Comment:
There are two code flows (non-overlaping) where this is set:
1. Here type is set to `TYPE_PRE_ALLOCATED` in case of ILB as well as a pre-set transaction id.
2. In SnapshotTrait, type is set to `TYPE_PRE_ALLOCATED ` only when a transaction id is provided.

Do you see any issues with this? I think we should follow one convention in both places.